backtesting/lib.py

```diff
@@ -452,6 +452,43 @@ def next(self):
                                self.data.Close[index] + self.__atr[index] * self.__n_atr)
 
 
+class PercentageTrailingStrategy(Strategy):
+    """
+    A strategy with automatic trailing stop-loss, trailing the current
+    price at distance of some percentage. Call
+    `PercentageTrailingStrategy.set_trailing_sl()` to set said percentage
+    (`5` by default). See [tutorials] for usage examples.
+
+    [tutorials]: index.html#tutorials
+
+    Remember to call `super().init()` and `super().next()` in your
+    overridden methods.
+    """
+    _sl_pct = 5.
+
+    def init(self):
+        super().init()
+
+    def set_trailing_sl(self, percentage: float = 5):
+        assert percentage > 0, "percentage must be greater than 0"
+        """
+        Sets the future trailing stop-loss as some (`percentage`)
+        percentage away from the current price.
+        """
+        self._sl_pct = percentage/100
+
+    def next(self):
+        super().next()
+        index = len(self.data)-1
+        for trade in self.trades:
+            if trade.is_long:
+                trade.sl = max(trade.sl or -np.inf,
+                               self.data.Close[index]*(1-self._sl_pct))
+            else:
+                trade.sl = min(trade.sl or np.inf,
+                               self.data.Close[index]*(1+self._sl_pct))
```

Comment:
```suggestion
        price = self.data.Close[-1]
        for trade in self.trades:
            if trade.is_long:
                trade.sl = max(trade.sl or -np.inf, price*(1 - self._sl_pct))
            else:
                trade.sl = min(trade.sl or np.inf, price*(1 + self._sl_pct))
```

Reply:
Before I make this change, I want to let you know that I used `index = len(self.data)-1` instead of -1 to be consistent with the TrailingStrategy code. I understand that in TrailingStrategy we were using ATR which was computed differently and to get an absolute value for an index, we had to use len(self.data)-1, But I don't see any harm in using it here. Thoughts?